includes/admin/class-wc-rest-payments-reporting-controller.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Class WC_REST_Payments_Reporting_Controller
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+use WCPay\Core\Server\Request\Get_Reporting_Payment_Activity;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller for customers.
+ */
+class WC_REST_Payments_Reporting_Controller extends WC_Payments_REST_Controller {
+
+	/**
+	 * Endpoint path.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'payments/reporting';
+
+	/**
+	 * Configure REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/payment_activity',
+			[
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_payment_activity' ],
+					'permission_callback' => [ $this, 'check_permission' ],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Retrieve the Payment Activity data.
+	 *
+	 * @param WP_REST_Request $request The request.
+	 * @return array
+	 */
+	public function get_payment_activity( $request ): array {
```

Comment:
Just to note that as I'm testing when server returns an error, this type hinting for the return value causes a problem because the return value is an object when server returns an error.

Looking at other controller, in client, we do not do type hinting in the client. Also, maybe because in the client, PHP version is not guaranteed to support type hinting.

Reply:
You are right, other controllers that use the `handle_rest_request` have avoided type hinting. Removed to follow the same for now to unblock.